tests/ext/aiohttp/test_client.py

```diff
@@ -0,0 +1,131 @@
+    async with ClientSession(loop=loop, trace_configs=[trace_config]) as session:
+        async with session.get(url):
+            pass
+
+    subsegment = xray_recorder.current_segment().subsegments[0]
+    assert subsegment.name == 'test'
+    assert subsegment.namespace == 'local'
+
+
+async def test_error(loop, recorder):
+    xray_recorder.begin_segment('name')
+    trace_config = aws_xray_trace_config()
+    status_code = 400
+    url = 'http://{}/status/{}'.format(BASE_URL, status_code)
+    async with ClientSession(loop=loop, trace_configs=[trace_config]) as session:
+        async with session.post(url):
+            pass
+
+    subsegment = xray_recorder.current_segment().subsegments[0]
+    assert subsegment.name == url
+    assert subsegment.error
+
+    http_meta = subsegment.http
+    assert http_meta['request']['url'] == url
+    assert http_meta['request']['method'] == 'POST'
+    assert http_meta['response']['status'] == status_code
+
+
+async def test_throttle(loop, recorder):
+    xray_recorder.begin_segment('name')
+    trace_config = aws_xray_trace_config()
+    status_code = 429
+    url = 'http://{}/status/{}'.format(BASE_URL, status_code)
+    async with ClientSession(loop=loop, trace_configs=[trace_config]) as session:
+        async with session.head(url):
+            pass
+
+    subsegment = xray_recorder.current_segment().subsegments[0]
+    assert subsegment.name == url
+    assert subsegment.error
+    assert subsegment.throttle
+
+    http_meta = subsegment.http
+    assert http_meta['request']['url'] == url
+    assert http_meta['request']['method'] == 'HEAD'
+    assert http_meta['response']['status'] == status_code
+
+
+async def test_fault(loop, recorder):
+    xray_recorder.begin_segment('name')
+    trace_config = aws_xray_trace_config()
+    status_code = 500
+    url = 'http://{}/status/{}'.format(BASE_URL, status_code)
+    async with ClientSession(loop=loop, trace_configs=[trace_config]) as session:
+        async with session.put(url):
+            pass
+
+    subsegment = xray_recorder.current_segment().subsegments[0]
+    assert subsegment.name == url
+    assert subsegment.fault
+
+    http_meta = subsegment.http
+    assert http_meta['request']['url'] == url
+    assert http_meta['request']['method'] == 'PUT'
+    assert http_meta['response']['status'] == status_code
+
+
+async def test_invalid_url(loop, recorder):
+    xray_recorder.begin_segment('name')
+    trace_config = aws_xray_trace_config()
+    async with ClientSession(loop=loop, trace_configs=[trace_config]) as session:
+        try:
+            async with session.get('http://doesnt.exist'):
+                pass
+        except Exception:
+            # prevent uncatch exception from breaking test run
+            pass
+
+    subsegment = xray_recorder.current_segment().subsegments[0]
+    assert subsegment.fault
+
+    exception = subsegment.cause['exceptions'][0]
+    assert exception.type == 'ClientConnectorError'
```

Comment:
This is where a `local` namespace makes sense. But it is optional to optimize `end_subsegment_with_exception` as it is not realistic to categorize every single error thrown by the library.

Reply:
Nops but tox uses 3.5.2 that is not compatible, aiohttp needs at least 3.5.3. Any suggestion?

Reply:
Forget about last comment. Its kinda weir that the place where the error comes from is handled by the value of the namespace. 

We can try to mark the most known exceptions as local and others as remote. In any case my doubt here is that taking into accout that the namespace is used to create the subsegment and the error is catched after the creation, can I modify the namespqce once is created?

Reply:
It's the other way around I believe: the namespace might need to be flipped back to `remote` depending on where the error comes from.

In this case the default captured subsegments should be `remote` as previously discussed. You can flip it back to `local` especially an error like this is thrown. After all non such endpoint could possibly exist so there won't be any downstream.